File: frontend-dev/src/type/projects.ts
export type Project = {
    id: number;
    title: string;
    slug: string;
    category: string;
    role: string;
    summary: string;
    image_url: string;
    github_url: string;
    demo_url: string;
    featured: boolean;
    display_order: number;
};

File: backend-dev/app/schemas/projects.py
from pydantic import BaseModel,ConfigDict
from datetime import datetime

class ProjectResponse(BaseModel):

    title: str
    slug: str
    category: str
    role: str
    summary: str
    image_url: str | None
    github_url: str | None
    demo_url: str | None
    featured: bool
    display_order: int
    created_at: datetime

    model_config = ConfigDict(from_attributes=True) #allows to build this schema from SQLAlchemy objects



